src/bin/cargo/commands/run.rs

```diff
@@ -93,13 +93,21 @@ pub fn is_manifest_command(arg: &str) -> bool {
         || path.file_name() == Some(OsStr::new("Cargo.toml"))
 }
 
-pub fn exec_manifest_command(config: &Config, cmd: &str, args: &[OsString]) -> CliResult {
+pub fn exec_manifest_command(config: &mut Config, cmd: &str, args: &[OsString]) -> CliResult {
     if !config.cli_unstable().script {
         return Err(anyhow::anyhow!("running `{cmd}` requires `-Zscript`").into());
     }
 
     let manifest_path = Path::new(cmd);
     let manifest_path = root_manifest(Some(manifest_path), config)?;
+
+    // Treat `cargo foo.rs` like `cargo install --path foo` and re-evaluate the config based on the
+    // environment where the script resides, rather than the environment from where its being run.
```

Comment:
```suggestion
    // location where the script resides, rather than the location from where it's being run.
```

Fix typo "its", and use "location" rather than the less specific "environment".